static_inverse_kinematics.py

```diff
@@ -60,7 +60,33 @@ def _marker_jac(self, q: np.ndarray, xp_markers: np.ndarray, n_frame: int):
 
         return jac
 
-    def solve(self, method: str = "lm"):
+    def optimize(self, n_frame: int, method: str, bounds: tuple() = None):
+        print(f" ****   Frame {n_frame}  ****")
+        x0 = self.q[:, n_frame - 1]
+        if bounds is None:
+            sol = scipy.optimize.least_squares(
+                    fun=self._marker_diff,
+                    args=([self.xp_markers[:, :, n_frame]]),
+                    jac=self._marker_jac,
+                    x0=x0,
+                    method=method,
+                    xtol=1e-6,
+                    tr_options=dict(disp=False),
+                )
+        else:
+            sol = scipy.optimize.least_squares(
+                fun=self._marker_diff,
+                args=([self.xp_markers[:, :, n_frame]]),
+                bounds=bounds,
+                jac=self._marker_jac,
+                x0=x0,
+                method=method,
+                xtol=1e-6,
+                tr_options=dict(disp=False),
+            )
+        self.q[:, n_frame] = sol.x
+
+    def solve(self, method: str = "lm", full: bool = False):
```

Comment:
What is the meaning of full = True with method = "trf"? If it has no sense, please create a new tag as a method which is "only_lm".

Reply:
done